src/test/java/org/wildfly/extension/elytron/SslTestCase.java

```diff
@@ -0,0 +1,137 @@
+package org.wildfly.extension.elytron;
+
+import org.jboss.as.subsystem.test.AbstractSubsystemTest;
+import org.jboss.as.subsystem.test.KernelServices;
+import org.jboss.msc.service.ServiceName;
+import org.junit.Assert;
+import org.junit.Test;
+
+import javax.net.ssl.SSLContext;
+import javax.net.ssl.SSLServerSocketFactory;
+import javax.net.ssl.SSLSocket;
+import javax.net.ssl.SSLSocketFactory;
+import javax.net.ssl.TrustManager;
+import javax.net.ssl.X509TrustManager;
+import java.net.InetSocketAddress;
+import java.net.ServerSocket;
+import java.security.cert.X509Certificate;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Executors;
+import java.util.concurrent.Future;
+
+/**
+ * @author <a href="mailto:jkalina@redhat.com">Jan Kalina</a>
+ */
+public class SslTestCase extends AbstractSubsystemTest {
+
+    private final int TESTING_PORT = 18201;
+
+    public SslTestCase() {
+        super(ElytronExtension.SUBSYSTEM_NAME, new ElytronExtension());
+    }
+
+    private KernelServices services = null;
+
+    private String keyStoreFile = SslTestCase.class.getResource("/org/wildfly/extension/elytron/testingCaJks.keystore").getFile();
```

Comment:
Is using the fully qualified name really necessary? By using the SslTestCase class to obtain the resource it should be possible to obtain the resource relative to that classes package already.